src/main/java/bio/terra/service/auth/iam/sam/SamIam.java

```diff
@@ -692,6 +692,11 @@ private void createGroupInner(String accessToken, String groupName) throws ApiEx
     samApiService.groupApi(accessToken).postGroup(groupName, null);
   }
 
+  @Override
+  public String getGroup(String accessToken, String groupName) throws InterruptedException {
+    return SamRetry.retry(configurationService, () -> getGroupEmail(accessToken, groupName));
```

Comment:
For at least one of the cases where we are calling getGroup, we expect it to fail in most cases. So, we may end up waiting a while for the retries to run. We may want a noRetry version of this call. What do you think?

Reply:
Hmm thats a good idea. Do you think I should make two versions of this method?

Reply:
Or, I could leave it as is because I know retry is implemented because there are frequent socket time out with Sam and we still want to account for those issues.

Reply:
> we expect it to fail in most cases

Would the failures we're expecting result in a retried API call? Looking at `SamRetry`, it doesn't appear to retry on 4xx errors, only 500 ones. In those cases, wouldn't we want to retry to make sure we're getting the correct response?

Reply:
Ah, those are good points. Leaving the retry makes sense.